### ISSUE TITLE
No check of certificate expiration date

### ISSUE THREAD

**Issue by JuergenKosel** [eclipse-paho/paho.mqtt.c#630]
Hello,

for compatibility tests of our application, we (a college of me and I) have tried to connect our MQTT client application to the MQTT broker of the IBM bluemix cloud.
We had accidently not updated the server certificate chain, so that the public certificate file has expired November 2017:
[messaging.pem.txt](https://github.com/eclipse/paho.mqtt.c/files/2969995/messaging.pem.txt)
To our surprise, the connection has been established without any warning or error message:
[log_extract.txt](https://github.com/eclipse/paho.mqtt.c/files/2969999/log_extract.txt)

*Question*: Is it the task of the application or should the paho.mqtt.c library check the expiration of the server certificate?

Greetings
  - Juergen

**Comment by icraggs** [comment]
The OpenSSL call to validate the server certificate is made in the client library: 

if (opts->enableServerCertAuth)
                        SSL_CTX_set_verify(net->ctx, SSL_VERIFY_PEER, NULL);

So I would think this ought to be checked for you.  You do need to set the TLS option enableServerCertAuth though.

**Comment by JuergenKosel** [comment]
Line 144 of the file log_extract.txt shows:
```
Mar 15 07:45:38 uagate014D OpcUa2MQTTPublisher[30432]: paho.mqtt.c: 20190315 074538.407 SSL certificate verification: X509_V_OK
Mar 15 07:45:38 uagate014D OpcUa2MQTTPublisher[30432]: paho.mqtt.c: 20190315 074538.407 SSL connect:SSL negotiation finished successfully
```
So it looks like that the certificate has passed the verification.

**Comment by icraggs** [comment]
Are you using the trustStore or CApath option to load the certificates?  The CApath (added for 1.3) option has an issue which means the certificates aren't loaded, which I'm fixing in the develop branch.  Could that be the problem?

**Comment by JuergenKosel** [comment]
Hello @icraggs ,

yes, this could be this problem.

**Comment by icraggs** [comment]
If you want to try it the fix, in SSLSocket.c, this code:

        if (opts->trustStore)
        {
                if ((rc = SSL_CTX_load_verify_locations(net->ctx, opts->trustStore, NULL)) != 1)
                {
                        if (opts->struct_version >= 3)
                                SSLSocket_error("SSL_CTX_load_verify_locations", NULL, net->socket, rc, opts->ssl_error_cb, opts->ssl_error_context);
                        else
                                SSLSocket_error("SSL_CTX_load_verify_locations", NULL, net->socket, rc, NULL, NULL);
                        goto free_ctx;
                }
        }

should become:

        if (opts->trustStore || opts->CApath)
        {
                if ((rc = SSL_CTX_load_verify_locations(net->ctx, opts->trustStore, opts->CApath)) != 1)
                {
                        if (opts->struct_version >= 3)
                                SSLSocket_error("SSL_CTX_load_verify_locations", NULL, net->socket, rc, opts->ssl_error_cb, opts->ssl_error_context);
                        else
                                SSLSocket_error("SSL_CTX_load_verify_locations", NULL, net->socket, rc, NULL, NULL);
                        goto free_ctx;
                }
        }

**Comment by icraggs** [comment]
The use of CApath was fixed in 1.3.1.  So hopefully this is fixed.
